src/main/java/org/spongepowered/api/entity/ai/task/AITask.java

```diff
@@ -72,10 +73,10 @@
      * 3. Returning "true" will add the provided task to the list of updated tasks, if not there
      *    already.
      *
-     * @param other The other task
+     * @param that The other task
      * @return True if it can be, false if not
      */
-    boolean canRunConcurrentWith(AITask<O> other);
+    boolean canRunConcurrentWith(AITask<?> that);
```

Comment:
Considering the situation that you are currently in
`LookIdleAITask extends AITask<Agent>` 
why wouldn't it be possible to execute 
`TargetAITask<A extends TargetAITask<A>> extends AITask<Creature>` concurrently?

````java
SwimmingAITask swim = ...;
FindNearestAttackableTargetAITask nearestAttackable = ...;
swim.canRunConcurrentWith(nearestAttackable); // Does not compile
````

Or is this fact silently cast away on the usage side? ([Yes it is](https://github.com/SpongePowered/SpongeCommon/blob/master/src/main/java/org/spongepowered/common/mixin/core/entity/ai/MixinEntityAITasks.java#L189))
If this fact is cast away silently then we can just use `<O>` anyway because we know, that we will only check tasks for the same entity/the owner of the task we currently use.
But `<?>` is the correct one if you ask me, why/where would you need the generic type here anyway?